packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -904,21 +904,34 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         const t: N.TsTypeAnnotation = this.startNode();
         this.expect(returnToken);
 
+        const assertsModifier = this.tsTryParse(
```

Comment:
Note that this could be parser without using `tryParse`: it would be possible to parse the `assert` identifier and then convert it to a type annotation if the next token is not another identifier.

On the other hand, we use `tryParse` everywhere when parsing TypeScript. I think that it's better to be consistent here, and then the whole TS parser can be optimized on other PRs.